src/main/java/org/folio/circulation/domain/policy/Period.java

```diff
@@ -216,6 +249,38 @@ public boolean isValid() {
     return duration != null && interval != null;
   }
 
+  public ZonedDateTime plusDate(ZonedDateTime date) {
+    switch (interval) {
+      case MONTHS:
+        return date.plusMonths(duration);
+      case WEEKS:
+        return date.plusWeeks(duration);
+      case DAYS:
+        return date.plusDays(duration);
+      case HOURS:
+        return date.plusHours(duration);
+      case MINUTES:
+      default:
+        return date.plusMinutes(duration);
+    }
+  }
```

Comment:
How can a date be added to a period?

Is this intended to advance a date by the given period?

Reply:
This is the code mentioned in the PR docs (In the Approach section) that I am not a fan of but is the quickest and simplest solution.
It seems that the `Period` can no longer be added to the date in the way it used to be done under JodaTime.

Rather than do more refactoring to create a better approach (which would result in even more changes), I opted to just have the Period accept the date to add itself to the date.

I would much rather be able to do `date.plusPeriod(period);`.

Reply:
> This is the code mentioned in the PR docs (In the Approach section) that I am not a fan of but is the quickest and simplest solution.

Ok, apologies I should've more thoroughly read that before commenting. I thought a quick skim might be valuable.

> It seems that the `Period` can no longer be added to the date in the way it used to be done under JodaTime.

Hmm, ok, given that `Period` is a circulation class, I wouldn't have expected the JodaTime classes to be able to directly handle it either.

> Rather than do more refactoring to create a better approach (which would result in even more changes), I opted to just have the Period accept the date to add itself to the date.
> 
> I would much rather be able to do `date.plusPeriod(period);`.

Sure. In order to do that we would need our own class for representing dates and times and that is out of scope of this work.

Is that correct?

Reply:
```
Sure. In order to do that we would need our own class for representing dates and times and that is out of scope of this work.

Is that correct?
```

Sort of.

In this case, I felt I was limited in time and could not achieve the goal of getting this done and solving this in a more correct way (not to mentioned trying to keep the changeset as small as reasonably possible).

I think this one can be improved if allotted more time to do so.
Such questions are beyond me as I am cycling off of this.

There is room for improvement and I think there is a good argument to improve it because the approach provided here effectively changes the style in ways that I too disagree with.

In short, this was a deadline decision rather than an out of scope decision.

To better solve this, I think changing the `private final String interval;` into a `ChronoUnit` and then go from there (such as dropping the `MONTHS`...and other constants, etc..).

Reply:
Just a follow up.
The reason why I would propose a `ChronoUnit` is because JavaTime can add arbitrary `ChronoUnit`'s if my memory serves me correctly.

I think a `ChronoUnit` can be used in a `TemporalAdjuster`, and there is a `with(TemporalAdjuster t)` method in ZonedDateTime.

Reply:
> In short, this was a deadline decision rather than an out of scope decision.

I don't think I understand the distinction, I do understand the intent.

> To better solve this, I think changing the private final String interval; into a ChronoUnit and then go from there (such as dropping the MONTHS...and other constants, etc..).

I agree that the periods internal state could use a ChronoUnit and that could be helpful in manipulating a date / time.

I'd like to get to a place where the domain can express this something like `date.advanceBy(period)`

That is a conversation for another day, as this isn't going to happen in this work.

Reply:
```
That is a conversation for another day, as this isn't going to happen in this work.
```
Then I guess we can conclude that the desired change is outside of scope and so we are left with the current approach taken by this PR.

Reply:
> Then I guess we can conclude that the desired change is outside of scope and so we are left with the current approach taken by this PR.

That is my interpretation of the statement you made above:

> In this case, I felt I was limited in time and could not achieve the goal of getting this done and solving this in a more correct way (not to mentioned trying to keep the changeset as small as reasonably possible).

Reply:
Done. I've added more detailed description to the `plusDate()` and `minusDate()` methods.

Reply:
@marc-johnson has this line of conversation been resolved to your satisfaction?